websites/S/Sonic Retro/presence.ts

```diff
@@ -0,0 +1,285 @@
+const presence = new Presence({
+		clientId: "970743721404530798"
+	}),
+	browsingTimestamp = Math.floor(Date.now() / 1000);
+presence.on("UpdateData", async () => {
+	const presenceData: PresenceData = {
+			startTimestamp: browsingTimestamp
+		},
+		{ pathname, search, hostname } = window.location;
+	switch (hostname) {
+		case "sonicretro.org": {
+			presenceData.smallImageKey = "home_512";
+			if (search) {
+				presenceData.largeImageKey = "search";
+				if (new URLSearchParams(document.location.search).get("s") === "")
+					presenceData.details = "Searching for something...";
+				else {
+					presenceData.details = `Searching for "${
+						document.querySelector("#main > div.archive-box > h1").textContent
+					}"`;
+				}
+			} else if (
+				((pathname === "/" || pathname === "/#") && !search) ||
+				(pathname.includes("/page/") && !search)
+			) {
+				presenceData.largeImageKey = "home";
+				presenceData.details = "Zooming through the main page";
+			} else if (pathname.includes("/category/")) {
+				presenceData.largeImageKey = "search";
+				presenceData.details = `Looking through ${
+					document.querySelector("#main > div.archive-box > h1").textContent
+				}`;
+			} else if (
+				/https:\/\/sonicretro.org\/\d{4}\/\d{2}\/(\d{2})*\//.test(
+					document.location.pathname
```

Comment:
now this just isn't smart
```suggestion
					pathname
```

Reply:
I dont even remember putting that, my bad boss